frontend/pages/policies/ManagePoliciesPage/components/CalendarEventsModal/CalendarEventsModal.tsx

```diff
@@ -133,28 +111,14 @@ const CalendarEventsModal = ({
     setFormData(newFormData);
   };
 
-  const onPolicyEnabledChange = useCallback(
-    (newVal: { name: FormNames; value: boolean }) => {
-      const { name, value } = newVal;
-      const newFormPolicies = formData.policies.map((formPolicy) => {
-        if (formPolicy.name === name) {
-          return { ...formPolicy, isChecked: value };
-        }
-        return formPolicy;
-      });
-      const newFormData = { ...formData, policies: newFormPolicies };
-      setFormData(newFormData);
-    },
-    [formData]
-  );
-
   const onUpdateCalendarEvents = () => {
     const errors = validateForm(formData);
 
     if (Object.keys(errors).length > 0) {
       setFormErrors(errors);
-    } else {
-      onSubmit(formData);
+    } else if (paginatedListRef.current) {
+      const changedPolicies = paginatedListRef.current.getDirtyItems();
+      onSubmit({ ...formData, changedPolicies });
     }
```

Comment:
The `PoliciesPaginatedList` component sends the list of changed items when it calls `onSubmit()`, but since this method (and others like it in the other modals) is also called when the enter key is pressed in the modal, we need another way to get the set of items, hence the call to `getDirtyItems()`.